test/test_optim.py

```diff
@@ -71,10 +71,10 @@ def test_adamw(self): self._test_adamw(1, {'lr': 0.001}, 1e-5, 0)
   def test_adamw_high_lr(self): self._test_adamw(1, {'lr': 10}, 1e-5, 1e-5)
 
   def test_multistep_adam(self): self._test_adam(10, {'lr': 0.001}, 1e-5, 0)
-  def test_multistep_adam_high_lr(self): self._test_adam(10, {'lr': 10}, 1e-5, 3e-4)
+  def test_multistep_adam_high_lr(self): self._test_adam(10, {'lr': 10}, 2e-4, 5e-4)
```

Comment:
Why are these worse? Does Adam use pow?

Reply:
There was a bit of discussion about it in the discord a while back. Its why the old pr for this was pretty complex https://github.com/geohot/tinygrad/pull/959

Reply:
Wait but the onnx pow isn't full two tensor pow. It's just like 0.9**<tensor>. This isn't accurate?

Reply:
Yea but it still goes through the approximation, and with higher lrs, the weights will be pretty big, and at large numbers it does start to diverge: https://godbolt.org/z/GfKPxcKTb

Reply:
Hmm, what is torch adam doing? pow is so slow.

Reply:
Yea it seems that they are: https://github.com/pytorch/pytorch/blob/main/torch/optim/adam.py#L366-L367

Reply:
Hmm, this is making me reconsider pow removal. I thought it wasn't really used, but if it's in Adam...

I still don't understand why the approximation is so bad. It has to be the inaccuracy in `exp2` itself right? Aka with a perfect `exp2` this would be perfect?

Reply:
Yea its probably the inaccuracy in exp2, but yea, with a perfect `exp2` it should be perfect, like if you just do `math.exp2(math.log2(0.9) * 50)` (needs 3.11 for exp2) it's way more accurate cuz its doubles instead of floats.